File: SDKs/parliament.uk/members/v1/python/src/sdk/models/operations/get_api_parties_getactive_house_.py
from dataclasses import dataclass, field
from typing import Optional
from sdk.models import shared


@dataclass
class GetAPIPartiesGetActiveHousePathParams:
    house: int = field(metadata={'path_param': { 'field_name': 'house', 'style': 'simple', 'explode': False }})
    

@dataclass
class GetAPIPartiesGetActiveHouseRequest:
    path_params: GetAPIPartiesGetActiveHousePathParams = field()
    

@dataclass
class GetAPIPartiesGetActiveHouseResponse:
    content_type: str = field()
    status_code: int = field()
    body: Optional[bytes] = field(default=None)
    party_members_service_search_result: Optional[shared.PartyMembersServiceSearchResult] = field(default=None)
    

File: SDKs/parliament.uk/members/v1/typescript/src/sdk/models/operations/getapipartiesgetactivehouse.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetApiPartiesGetActiveHousePathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=house" })
  house: number;
}


export class GetApiPartiesGetActiveHouseRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetApiPartiesGetActiveHousePathParams;
}


export class GetApiPartiesGetActiveHouseResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  body?: Uint8Array;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  partyMembersServiceSearchResult?: shared.PartyMembersServiceSearchResult;

  @SpeakeasyMetadata()
  statusCode: number;
}
